### PR TITLE
refactoring user profile connections/followers for fewer requests, shared code

### PR DESCRIPTION
**Pull request by 2is10**
This puts the list of ids in the original response for the first page of connections/followers.
It also combines the endpoints for loading more connections or followers by ID.
@ahsu1230

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala [hunk 1]
```diff
@@ -71,132 +72,99 @@ class UserController @Inject() (
     libraryMembershipRepo: LibraryMembershipRepo,
     fortytwoConfig: FortyTwoConfig) extends UserActions with ShoeboxServiceController {
 
-  private def loadConnectionUsers(userIds: Set[Id[User]], viewerIdOpt: Option[Id[User]], friendIds: Set[Id[User]] = Set.empty) = {
```
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala:74]: Passing `friendStatusCommander.augmentUsers` an empty set of friend IDs would be incorrect if the user is in fact friends with some of the users. There’s another method by the same name that doesn’t take a set of friend IDs, and it looks them up. That’s the one we need to call when we don’t have a friend ID set handy. Sorry, I suppose the `FriendStatusCommander` API could be clearer.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala [hunk 2]
```diff
@@ -71,132 +72,99 @@ class UserController @Inject() (
...
-      //mutual
-      viewerIdOpt.map { viewerId =>
-        val followingLibCount = libraryRepo.countLibrariesOfOwnerUserFollow(userId, viewerId) //not cached
-        val mutualConnectionCount = userConnectionRepo.getMutualConnectionCount(userId, viewerId) //cached
-        jsonWithGlobalCounts +
-          ("mConnections" -> JsNumber(mutualConnectionCount)) +
-          ("mLibraries" -> JsNumber(followingLibCount))
-      } getOrElse jsonWithGlobalCounts
-    }
-  }
```
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala:109]: I didn’t change this method at all except to use an existing DB connection. You can view the diff with `?w=` in the location bar to verify.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala (hunk 1 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala:74]: Ah, I see - sorry, was confused =D

**Comment by ahsu1230** [comment]
Thanks Jared, I think it looks good.
I think we'll probably want to start thinking about how we should cache follower counts or followers, which might be a bit trickier since it depends on the viewer?

**Comment by ahsu1230** [comment]
also what do you think about putting these profile endpoints into a `ProfileController`?
I think `UserController` makes sense to only contain user settings and preferences while `LibraryController` has more specific library actions

`ProfileController` often depends on who's viewing (anonymous, friend, owner, etc.). For followers & libraries at least, it contains similar logic.
Just a thought for when things settle down and we have time to refactor.

**Comment by 2is10** [comment]
> also what do you think about putting these profile endpoints into a ProfileController?

I had the same idea today. Great minds think alike. :smile:

**Pull request merged**